src/hobbits-plugins/operators/Edit/edit.cpp

```diff
@@ -0,0 +1,236 @@
+    return newBits;
+}
+
+QSharedPointer<BitArray> Edit::parseHex(QString newBitsInRange) 
+{
+    QStringList parseErrors = QStringList();
+    QSharedPointer<BitArray> newBitArray;
+    QString bitArraySpec = "0x"+newBitsInRange;
+    QSharedPointer<BitArray> newBits = newBitArray->fromString(bitArraySpec, &parseErrors);
+    return newBits;
+}
+
+QSharedPointer<BitArray> Edit::parseAscii(QString newBitsInRange) {
+        QStringList parseErrors = QStringList();
+        QSharedPointer<BitArray> newBitArray;
+        QString bitArraySpec = newBitsInRange;
+        QSharedPointer<BitArray> newBits = newBitArray->fromString(bitArraySpec, &parseErrors);
+        return newBits;
+
+        /*
+
+        QByteArray nair = newBitsInRange.toLocal8Bit();
+        const char * newAsciiInRange = nair.data();
+
+        int newBitsInRangeLength = newBitsInRange.length();
+        qint64 outputSize = inputContainerSize - (length * 8) + (newBitsInRangeLength * 8);
+
+        QSharedPointer<const BitArray> bits = inputContainers.at(0)->bits();
+
+        if (newBitsInRangeLength == length) {     
+            m_outBits->setBytes(start, newAsciiInRange, 0, newBitsInRangeLength);
+        } else {
+            QSharedPointer<BitArray> postBits = QSharedPointer<BitArray>(new BitArray(outputSize - start + length ));
+            QByteArray byteArray = bits->readBytes(start + length, outputSize - 1);
+            int endLength = byteArray.length();
+            const char* data = byteArray.data();
+            m_outBits->setBytes(start+newBitsInRangeLength, data, 0, endLength);
+            m_outBits->setBytes(start, newAsciiInRange, 0, newBitsInRangeLength);
+            
+        }
+
+        */
+
+}
+
+QSharedPointer<const OperatorResult> Edit::operateOnBits(
+    QList<QSharedPointer<const BitContainer> > inputContainers,
+    const Parameters &parameters,
+    QSharedPointer<PluginActionProgress> progress)
+{    
+    if (inputContainers.length() == 0) {
+        QList<QSharedPointer<BitContainer>> outputContainers;
+        qint64 outputSize = 10;
+        QSharedPointer<BitArray> outBits;
+        outBits = QSharedPointer<BitArray>(new BitArray(outputSize));
+
+        auto bitContainer = BitContainer::create(outBits);
+
+        outputContainers.append(bitContainer);
+        return OperatorResult::result(outputContainers, parameters);
+    }
+
+    if (inputContainers.length() > 1) {
+        return OperatorResult::error("Requires a single input bit container");
+        
+    }
+
+    QStringList invalidations = m_delegate->validate(parameters);
+    if (!invalidations.isEmpty()) {
+        return OperatorResult::error(QString("Invalid parameters passed to %1:\n%2").arg(name()).arg(invalidations.join("\n")));
+    }
+
+    int progressInt = 15;
+
+    progress->setProgressPercent(progressInt);
+
+    //get values from spin boxes
+    qint64 start = parameters.value("start").toInt();
```

Comment:
these comments are outdated - you generally shouldn't be referencing the GUI in this class